source/common/router/config_impl.cc

```diff
@@ -484,6 +498,8 @@ RouteMatcher::RouteMatcher(const Json::Object& json_config, const ConfigImpl& gl
           throw EnvoyException(fmt::format("Only a single single wildcard domain is permitted"));
         }
         default_virtual_host_ = virtual_host;
+      } else if ('*' == domain[0]) {
+        wildcard_virtual_host_suffixes_[domain.size() - 1].emplace(domain.substr(1), virtual_host);
```

Comment:
It looks like the schema allows it to be an empty string. Since the danger is at runtime and non-debug binaries have ASSERTs compiled out, I'll add an explicit check.